build.gradle

```diff
@@ -1,15 +1,15 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-  ext.kotlin_version = '1.3.41'
+  ext.kotlin_version = '1.4.21'
   ext.fragment_version = '1.2.0-rc01'
   repositories {
     google()
     jcenter()
     gradlePluginPortal()
   }
   dependencies {
-    classpath 'com.android.tools.build:gradle:3.6.1'
+    classpath 'com.android.tools.build:gradle:4.1.1'
```

Comment:
@anandwana001 Only this is required for the issue that contributors might be facing with AS 4.2 + / Canary version.

Reply:
What issue is being encountered, and can it be avoided by not using canary?

Reply:
As an FYI, I think we can still use 3.6.1 (Konfetti looks like it's still working when I just ran it on the emulator), what are your thoughts @BenHenning?

Reply:
@BenHenning it can be avoided if not using Canary but once the official update for 4.2 is out build would be breaking out as there is some issue with data binding and AS 4.2 and above so we would eventually updating this. @jcqli bdw we can use a simpler version of Konfetti is we don't want a bunch of customization.

Reply:
So there are two options then, using a simpler version of Konfetti or keeping Gradle 3.6
It looks like using the latest Konfetti and Kotlin 1.4 is still compatible with Gradle 3.6, and this is what I've just pushed
@aggarwalpulkit596 Does this work with Canary and 4.2?

Reply:
@aggarwalpulkit596 what's the specific issue? Is it only when using databinding + AS 4.2 + Kotlin 1.4? I'm trying to understand the parameters, the specific issue, and the broader context. Do you know whether the Android Studio team is aware of the issue? Is it Gradle-specific? Do we have an issue with any additional context?

Reply:
As an update, since we can more easily use an earlier version of Konfetti we'll keep Kotlin 1.3 (there were several errors that came up when trying to migrate to 1.4 and ensure compatibility with Gradle and other dependencies).
So currently it has Kotlin 1.3, Gradle plugin 3.6, and Gradle wrapper 6.5 (instead of 5.6 that we were using earlier, based on [this](https://docs.gradle.org/6.5/userguide/compatibility.html) compatibility).

How does this sound?

Reply:
I think it might be an AS caches issue. If others encounter this, can you try to invalidate caches & restart after applying Jacque's changes to see if the error goes away? (mostly @aggarwalpulkit596 since you brought this up originally :) )

Reply:
@BenHenning the issue wasn't related to kotlin 1.4 it was only related to databinding + AS 4.2 and yes it was an official issue on the issue tracker that's how i got to know about the problem in the first place.

Reply:
Thanks @aggarwalpulkit596. Can you add a link to that? It sounds like something we don't to block this PR on.